internal/proposals/logs/improvement-of-log-messages-usability.md

```diff
@@ -0,0 +1,92 @@
+# Improvement of log messages usability
+
+## Motivation
+
+We want to provide a common, unified solution to logging across Kyma components, which will enable users and maintainers to easily track specific events happening in Kyma's ecosystem. The solution should be independent of the platform (AWS, Azure, GCP) and easy to introduce in existing components
+
+Users, developers, and maintainers want to have an easy way to track events happening in Kyma, problems that occur, and debug faulty components. This will make understanding of Kyma's ecosystem easier and more transparent
+
+## Solution regarding the log messages
+
+Every component should produce log entries in order to make the problem diagnosis process quick and effective
+
+Key assumptions for the log messages:
+
+- Transparency - every entry must be as short as possible but containing all essential information about the process
+- Tracing - entries sequence must be connected so there is an easy way to filter logs concerning a specific operation
+- GDPR-compliance - entries must not contain any sensitive data (password, personal data in the meaning of GDPR)
+- No "responsibility overlapping" - Because there are other tools to help in diagnosis, logs, in general, should not contain the information which can be taken from that tools (i.e there's tracing which can be used for timings)
+
+Additional assumptions:
+
+- If there is one quick solution for an occurring issue, add it to the message on the DEBUG level so the operator could use it while debugging
+- If operation will be retried, say when
+- JSON format should be enabled only on environments where the monitoring is enabled. Text format is more readable for debugging purposes so make it configurable
+- In case of debug level add the error stack tracing
+- Don't put redundant data into the logs. If something failed 10 times, log that it failed 10 times instead of printing the same error 10 times
+
+## Log structure
+
+To unify the logs, therefore to make debugging process much easier and to make logs parsing also super easy, I'd like to propose a single log format so every service, job and all the components could follow:
+
+- timestamp - RFC3339 Date and time with timezone. For example: "2012-12-12T07:20:50.52Z"
```

Comment:
```suggestion
- Timestamp - RFC3339 format of date, time, and timezone. For example: "2012-12-12T07:20:50.52Z".
```

Reply:
If these are the names of fields, I'd leave them lowercase and format them instead, for example with **bold** formatting.

Reply:
Ok! I'll do that